x-pack/legacy/plugins/integrations_manager/server/plugin.ts

```diff
@@ -33,19 +33,20 @@ export class Plugin {
   public setup(core: CoreSetup) {
     const { http, elasticsearch } = core;
     const { server } = http;
-
-    // make these items available to handlers via h.context
-    // https://github.com/hapijs/hapi/blob/master/API.md#server.bind()
-    // aligns closely with approach proposed in handler RFC
-    // https://github.com/epixa/kibana/blob/rfc-handlers/rfcs/text/0003_handler_interface.md
     const context: PluginContext = {
       esClient: elasticsearch.createClient(PLUGIN_ID),
     };
 
-    server.bind(context);
-
     // map routes to handlers
-    routes.forEach(route => server.route(route));
+    routes.forEach(route => {
+      // make these items available to handlers via h.context
+      // https://github.com/hapijs/hapi/blob/master/API.md#route.options.bind
+      // aligns closely with approach proposed in handler RFC
+      // https://github.com/epixa/kibana/blob/rfc-handlers/rfcs/text/0003_handler_interface.md
+      if (!route.options) route.options = {};
+      if (!route.options.bind) route.options.bind = context;
```

Comment:
:+1: For binding to the route options instead of `server.bind`

I don't understand the if statements / logic. Wouldn't your routes completely break when `if (!route.options.bind)` evaluates to false and these routes no longer receive the correct context object?

Reply:
You're reading it right. I wanted to respect any value a route supplied, even if it lost/clobbered the correct context, but the existing code ignores it entirely.

Updated in 383ef4975301990c6f1122762ba3190d396d401a to merge route.options.bind into the plugin context